torch/csrc/autograd/functions/tensor.cpp

```diff
@@ -79,8 +79,8 @@ auto CopySlices::apply(variable_list&& inputs) -> variable_list {
   result.copy_(grad);
 
   at::Tensor grad_slice;
-  if (view_fn.has_value()) {
-    auto fn = view_fn.value();
+  if (view_fn) {
+    auto fn = view_fn;
```

Comment:
Is this still needed now that it is not an optional anymore?

Reply:
Yes, it's needed as `view_fn` can be `nullptr` for CopySlices.
As the [comment](https://github.com/pytorch/pytorch/blob/eee849be8cc6cce5ce867afb7d95aec521b16dce/torch/csrc/autograd/functions/tensor.h#L60-L67) shown, on CPU/GPU, we can use TensorGeometry of both base and view tensors rather than `view_fn` to create CopySlices.
https://github.com/pytorch/pytorch/pull/50067#discussion_r552211994

Reply:
Right but why not directly use `view_fn` in the call below? This shallow copy of the function isn't doing anything here?

Reply:
I see. You are right. I should directly call `view_fn`.